client/webserver/webserver.go

```diff
@@ -261,6 +263,7 @@ func New(core clientCore, addr, customSiteDir string, logger dex.Logger, reloadH
 	mux.Route("/api", func(r chi.Router) {
 		r.Use(middleware.AllowContentType("application/json"))
 		r.Post("/init", s.apiInit)
+		r.Post("/shutdown", s.apiShutdown)
```

Comment:
Hmm. Seems like anyone can shutdown our client from anywhere at any time. 

I seriously question whether this should exist as an API endpoint at all. For the specific case of Decrediton integration, would it be better to add a function to `libdexc` `preInitMethods` to handle context cancellation? Or is there another case for a general purpose API endpoint to shut down the client?

Reply:
This certainly should be behind the auth middleware.

This is also not applicable to Decrediton integration anymore, but I don't see a fundamental problem with controlling the full lifetime of dexc programmatically.  Signals don't work because we catch them with a prompt.  A different signal could skip the prompt, but that's getting very OS specific.

Reply:
We could consider eliminating the terminal prompt as a resolution, but that makes me pretty uncomfortable too knowing how people use software.